### PR TITLE
manage the async stuff outside of the dropdown

### PR DESCRIPTION
**Pull request by patelvarun**

### CODE REVIEW

File: packages/base/components/Input/Dropdown.js [hunk 1]
```diff
@@ -76,11 +100,18 @@ const Dropdown = ({ input, ...rest }) => {
                 {...inputRest}
                 {...input}
                 backspaceRemoves={backspaceRemoves}
-                cacheOptions
+                cacheOptions={cacheOptions}
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: I think we can just remove this props & also remove it from above so it's automatically passed in `inputRest`.

File: packages/base/components/Input/Dropdown.js [hunk 2]
```diff
@@ -64,8 +83,13 @@ const Dropdown = ({ input, ...rest }) => {
               input.onChange(parsedValue, meta);
             };
 
-            const optionByValue = value =>
-              (inputRest.options || asyncOptions).find(o => o.value === value);
+            const optionByValue = value => {
+              // After search add new options in state
+              const newAsyncOptions = rest.asyncPrimaryOptions || asyncOptions;
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: I think we want to destructure `asyncPrimaryOptions` & `setAsyncPrimaryOptions` right? I don't think they are valid `react-select` props so we shouldn't pass them through. They should also be added to `propTypes`.

**Comment by patelvarun** [comment]
> Thanks for the changes. Couple requests. Also, can you share an example of how this is used? Need to refresh my memory on the context here.
=> *Below is our chat:*
Varun:
Async dropdown value is not getting selected (updated) because of `asyncOptions` in dropdown in create/update schema page. Below is the code where `value` is updated after the drop but we have old `asyncOptions` in the state. so it is not matching and not getting selected.

const optionByValue = value =>
             (inputRest.options || asyncOptions).find(o => o.value === value);
Cade:
I've wrapped react-select such that the value is the actual selected value instead of an object with label/value.
when we swap/move the field array values, we are _only_ swapping the values (from what I understand), and not the actual component—meaning the component state doesn't change, as you discovered. 

So to solve this I used async stuff outside of the dropdown component.

File: packages/base/package.json [hunk 3]
```diff
@@ -33,6 +33,7 @@
     "aws-appsync-react": "^1.2.8",
     "deep-object-diff": "^1.1.0",
     "isomorphic-unfetch": "^3.0.0",
+    "lodash": "^4.17.11",
```
Comment (cadejscroggins) [inline packages/base/package.json]: Should be a peer dependency.

File: packages/base/components/Input/Dropdown.js [hunk 4]
```diff
@@ -3,17 +3,19 @@ import React, { useState } from 'react';
...
+import uniqBy from 'lodash/uniqBy';
 import InputWrapper from './InputWrapper';
 
 const Dropdown = ({ input, ...rest }) => {
   const [asyncOptions, setAsyncOptions] = useState([]);
-
   return (
     <InputWrapper input={input} {...rest}>
       {({
+        asyncPrimaryOptions = null,
```
Comment (cadejscroggins) [inline packages/base/components/Input/Dropdown.js]: Now that I understand this, let's rename these to `externalAsyncOptions` & `setExternalAsyncOptions`.

**Pull request merged**